core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2078,6 +2078,12 @@ public void delete(AlluxioURI path, DeleteContext context)
         }
 
         deleteInternal(rpcContext, inodePath, context, false);
+        if (context.getOptions().getAlluxioOnly()
+            && context.getOptions().hasResetDirectChildrenLoadedState()
```

Comment:
if this field is not set, you'll get a false so you don't need to check hasResetDirectChildrenLoadedState specifically.

Reply:
If this field is not set by client but we call `getResetDirectChildrenLoadedState()`, an NPE will ocurr.